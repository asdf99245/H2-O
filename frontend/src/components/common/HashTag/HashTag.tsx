import styled from '@emotion/styled';
import { Typography } from '@/components/common';

interface Props {
  title: string;
}
function HashTag({ title }: Props) {
  return (
    <StyleHashTag as='span' font='TextKRRegular12' color='gray50'>
      {title}
    </StyleHashTag>
  );
}

export default HashTag;

const StyleHashTag = styled(Typography)`
  align-items: center;
  background-color: ${({ theme }) => theme.colors.hashTagBg};
  border-radius: 2px;
  padding: 2px 6px;
  width: fit-content;
  height: fit-content;
`;
