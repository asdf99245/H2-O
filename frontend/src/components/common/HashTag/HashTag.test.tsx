import HashTag from './HashTag';
import { render, screen } from '@/tests/test-util';

describe('HashTag 컴포넌트 테스트', () => {
  it('HashTag가 정상적으로 렌더링된다.', () => {
    render(<HashTag title='해시태그' />);

    expect(screen.getByText('해시태그')).toBeInTheDocument();
  });

  it('HashTag 스냅샷 테스트', () => {
    render(<HashTag title='해시태그' />);

    expect(screen.getByText('해시태그')).toMatchSnapshot();
  });
});
