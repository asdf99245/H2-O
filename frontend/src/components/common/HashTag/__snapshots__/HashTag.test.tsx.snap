// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HashTag 컴포넌트 테스트 > HashTag 스냅샷 테스트 1`] = `
.emotion-0 {
  color: #F7F8F9;
  font-family: Hyundai Sans Text KR;
  font-size: 12px;
  font-weight: 400;
  line-height: 18px;
  letter-spacing: -0.03em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #75757580;
  border-radius: 2px;
  padding: 2px 6px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

<span
  class="emotion-0"
  color="gray50"
>
  해시태그
</span>
`;
